streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```diff
@@ -450,7 +438,7 @@ public Headers headers() {
     @Override
     public long timestamp() {
         if (recordTimestamp == null) {
-            throw new IllegalStateException("Timestamp must be set before use via setRecordMetadata() or setTimestamp().");
```

Comment:
I would say that instead of removing this part we should mention the non deprecated method `setRecordTimestamp`

Reply:
I second this.